src/sagas/events.js

```diff
@@ -14,22 +14,38 @@ import { refreshEvents } from '../actions';
 
 function* createEvent({ eventData }) {
 
+    // let options = {
+    //     enableHighAccuracy: true,
+    //     timeout: 1000,
+    //     maximumAge: 0
+    // };
+
     const updates = {};
     const eventUid = '2018/03/'+ Math.round(Math.random()*1000000);
     updates['events/' + eventUid] = {
         ...eventData,
         // add timestamp?
     };
 
-    window._FIREBASE_DB_.ref()
-        .update(updates);
+    navigator.geolocation.getCurrentPosition(
```

Comment:
Can you glob the lat/long onto the eventData instead of a separate read/write? That way even if lat/long fails, the event can still be logged?

```javascript

let latLong = {};

latLong = navigator.getCurrentPosition();

latLong.then( latLong => {
updates['events/' + eventUid] = {
         ...eventData,
         ...latLong,
         // add timestamp?
     };	 
})
.catch( latLongErr => {
updates['events/' + eventUid] = {
         ...eventData,
         // add timestamp?
     };	 
})

```

Reply:
Yeah, this is my second comment.  We just go along silently if they opt out of geolocation.  @ahulab question remains, you think accuracy is required?  Just log everything for now?